sapi/cli/php_cli_server.c

```diff
@@ -2241,20 +2241,13 @@ static bool php_cli_server_dispatch_router(php_cli_server *server, php_cli_serve
 
 	zend_try {
 		zval retval;
-
-		/* Normally php_execute_script restarts the timer with max_execution_time if it has
-		 * previously been initialized with max_input_time. We're not using php_execute_script here
-		 * because it does not provide a way to get the return value of the main script, so we need
-		 * to restart the timer manually. */
-		if (PG(max_input_time) != -1) {
-#ifdef PHP_WIN32
-			zend_unset_timeout();
-#endif
-			zend_set_timeout(INI_INT("max_execution_time"), 0);
-		}
-
 		ZVAL_UNDEF(&retval);
-		if (SUCCESS == zend_execute_scripts(ZEND_REQUIRE, &retval, 1, &zfd)) {
+		int sg_options_back = SG(options);
+		/* Don't chdir to the router script because the file path may be relative. */
+		SG(options) |= SAPI_OPTION_NO_CHDIR;
```

Comment:
This looks suspicious.
Why did we change ``cwd`` before and decided not to change it now? Is this documented somewhere?
In any case this is not related to "syntax error message fix".

Reply:
What this essentially does is switching from `zend_execute_scripts` to `php_execute_script`. `php_execute_script` handles certain things (like error messages, and timeouts) which `php_execute_script` does not. However, it also switches `cwd`. To avoid this break, the file is executed without `SAPI_OPTION_NO_CHDIR`.

Another solution would be to resolve the script path to an absolute one. I think that should work as well.

Reply:
I missed the usage ``SAPI_OPTION_NO_CHDIR`` usage in ``php_execute_script()``. My mistake.
I don't see any other problems.